local-setup/scripts/ci-dataloader-v2-regression.py

```
@@ -0,0 +1,696 @@
+    # Login as that user
+    test_loader = CRSLoader(BASE_URL)
+    logged_in = test_loader.login(
+        username="CI-NOBND", password="eGov@123", tenant_id=test_tenant
+    )
+    if not logged_in:
+        # If login fails (e.g., HRMS bug), skip this test gracefully
+        print("   ⚠️ User login failed (known HRMS bug), testing with mock 403 instead")
+        # Test the error detection code path directly
+        from unified_loader import APIUploader
+        uploader = APIUploader(BASE_URL)
+        uploader._auth_token = "invalid-token-for-403"
```

Ruff (0.15.9)

[error] Possible hardcoded password assigned to: "_auth_token"

(S105)